sdks/go/pkg/beam/io/rtrackers/offsetrange/offsetrange_test.go

```diff
@@ -299,3 +301,299 @@ func TestTracker_TrySplit(t *testing.T) {
+			claimed:           5,
+			fraction:          -0.5,
+			rangeEndEstimator: offsetRangeEndEstimator{EstimateRangeEnd: 10},
+			splitPt:           5,
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(fmt.Sprintf("(split at %v of [%v, %v])",
+			test.fraction, test.claimed, test.rest.End), func(t *testing.T) {
+			rt, err := NewGrowableTracker(test.rest, &test.rangeEndEstimator)
+			if err != nil {
+				t.Fatalf("error in creating a new growable tracker: %v", err)
+			}
+			ok := rt.TryClaim(test.claimed)
+			if !ok {
+				t.Fatalf("tracker failed on initial claim: %v", test.claimed)
+			}
+			gotP, gotR, err := rt.TrySplit(test.fraction)
+			if err != nil {
+				t.Fatalf("tracker failed on split: %v", err)
+			}
+			var wantP interface{} = Restriction{Start: test.rest.Start, End: test.splitPt}
+			var wantR interface{} = Restriction{Start: test.splitPt, End: test.rest.End}
+			if test.splitPt == test.rest.End {
+				wantR = nil // When residuals are empty we should get nil.
+			}
+			if !cmp.Equal(gotP, wantP) {
+				t.Errorf("split got incorrect primary: got: %v, want: %v", gotP, wantP)
+			}
+			if !cmp.Equal(gotR, wantR) {
+				t.Errorf("split got incorrect residual: got: %v, want: %v", gotR, wantR)
+			}
+		})
+	}
+}
+
+// TestGrowableTracker_IsBounded tests IsBounded method for GrowableTracker.
+func TestGrowableTracker_IsBounded(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+
+	if tracker.IsBounded() {
+		t.Errorf("GrowableTracker is bounded, want unbounded initially.")
+	}
+
+	if !tracker.TryClaim(int64(0)) {
+		t.Errorf("tracker.TryClaim(0) = %v, want: %v", false, true)
+	}
+
+	if tracker.IsBounded() {
+		t.Errorf("GrowableTracker should've been unbounded.")
+	}
+
+	estimator.SetEstimateRangeEnd(16)
+	tracker.TrySplit(0.5)
+	if !tracker.IsBounded() {
+		t.Errorf("tracker should've been bounded after split")
+	}
+}
+
+// TestGrowableTracker_Progress tests GetProgess method for GrowableTracker.
+func TestGrowableTracker_Progress(t *testing.T) {
+	var start, cur int64
+	start = 10
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: start, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error in creating a new GrowableTracker: %v", err)
+	}
+	cur = 20
+
+	if !tracker.TryClaim(int64(cur)) {
+		t.Fatalf("couldn't claim tracker.TryClaim(int64(%v))", cur)
+	}
+
+	estimator.SetEstimateRangeEnd(5)
+	done, remaining := tracker.GetProgress()
+	if float64(cur+1-start) != done {
+		t.Errorf("wrong amount of work done, got: %v, want: %v", cur+1-start, done)
+	}
+	if math.MaxInt64-done != remaining {
+		t.Errorf("work remaining:%v, expected: %v", remaining, math.MaxInt64-done)
+	}
```

Comment:
To avoid repeating the math in the message, assign them to `got` and `want`, and avoid the possibility of error entirely. (which saves me, as a reviewer, from having to check)

eg
```
	if got, want := done, float64(cur+1-start); got != want {
		t.Errorf("wrong amount of work done, got: %v, want: %v", got, want)
	}